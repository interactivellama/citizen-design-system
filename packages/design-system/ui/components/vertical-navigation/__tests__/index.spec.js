// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { mount } from 'enzyme';
import {
  NavVerticalRadioSet,
  NavVerticalFieldset,
  NavVerticalItemRadio
} from '../radio-group/example';

const { matchesMarkup } = createHelpers(__dirname);

describe('Vertical Nav Radio Group', () => {
  it('renders a focus state', () => {
    const wrapper = mount(<NavVerticalRadioSet />);
    return matchesMarkup(
      wrapper
        .find('.slds-nav-vertical__item')
        .at(0)
        .simulate('focus')
        .html()
    );
  });

  it('renders a hover state', () => {
    const wrapper = mount(<NavVerticalRadioSet />);
    return matchesMarkup(
      wrapper
        .find('.slds-nav-vertical__item')
        .at(0)
        .simulate('mouseEnter')
        .html()
    );
  });

  it('renders a radio button', () =>
    matchesMarkup(<NavVerticalItemRadio id="test1" />));

  it('renders a checked radio button', () =>
    matchesMarkup(<NavVerticalItemRadio id="test1" checked />));

  it('renders a fieldset', () =>
    matchesMarkup(
      <NavVerticalFieldset>
        <NavVerticalItemRadio id="test1">test1</NavVerticalItemRadio>
      </NavVerticalFieldset>
    ));

  it('renders a shaded fieldset', () =>
    matchesMarkup(
      <NavVerticalFieldset shaded>
        <NavVerticalItemRadio id="test1">test1</NavVerticalItemRadio>
      </NavVerticalFieldset>
    ));

  it('renders a compact fieldset', () =>
    matchesMarkup(
      <NavVerticalFieldset compact>
        <NavVerticalItemRadio id="test1">test1</NavVerticalItemRadio>
      </NavVerticalFieldset>
    ));
});
