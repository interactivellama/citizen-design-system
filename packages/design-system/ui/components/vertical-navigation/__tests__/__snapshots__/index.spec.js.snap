// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Nav Radio Group renders a checked radio button 1`] = `
<span
  className="slds-nav-vertical__item"
>
  <input
    defaultChecked={true}
    id="test1"
    name="example"
    type="radio"
    value="test1"
  />
  <label
    className="slds-nav-vertical__action"
    htmlFor="test1"
  >
    <span
      className="slds-nav-vertical_radio-faux"
    />
  </label>
</span>
`;

exports[`Vertical Nav Radio Group renders a compact fieldset 1`] = `
<fieldset
  className="slds-nav-vertical slds-nav-vertical_compact"
>
  <legend
    className="slds-nav-vertical__title slds-text-title_caps"
  >
    Reports
  </legend>
  <span
    className="slds-nav-vertical__item"
  >
    <input
      defaultChecked={undefined}
      id="test1"
      name="example"
      type="radio"
      value="test1"
    />
    <label
      className="slds-nav-vertical__action"
      htmlFor="test1"
    >
      <span
        className="slds-nav-vertical_radio-faux"
      >
        test1
      </span>
    </label>
  </span>
</fieldset>
`;

exports[`Vertical Nav Radio Group renders a fieldset 1`] = `
<fieldset
  className="slds-nav-vertical"
>
  <legend
    className="slds-nav-vertical__title slds-text-title_caps"
  >
    Reports
  </legend>
  <span
    className="slds-nav-vertical__item"
  >
    <input
      defaultChecked={undefined}
      id="test1"
      name="example"
      type="radio"
      value="test1"
    />
    <label
      className="slds-nav-vertical__action"
      htmlFor="test1"
    >
      <span
        className="slds-nav-vertical_radio-faux"
      >
        test1
      </span>
    </label>
  </span>
</fieldset>
`;

exports[`Vertical Nav Radio Group renders a focus state 1`] = `"<span class=\\"slds-nav-vertical__item\\"><input type=\\"radio\\" id=\\"recent\\" value=\\"recent\\" name=\\"example\\"><label class=\\"slds-nav-vertical__action\\" for=\\"recent\\"><span class=\\"slds-nav-vertical_radio-faux\\">Recent</span></label></span>"`;

exports[`Vertical Nav Radio Group renders a hover state 1`] = `"<span class=\\"slds-nav-vertical__item\\"><input type=\\"radio\\" id=\\"recent\\" value=\\"recent\\" name=\\"example\\"><label class=\\"slds-nav-vertical__action\\" for=\\"recent\\"><span class=\\"slds-nav-vertical_radio-faux\\">Recent</span></label></span>"`;

exports[`Vertical Nav Radio Group renders a radio button 1`] = `
<span
  className="slds-nav-vertical__item"
>
  <input
    defaultChecked={undefined}
    id="test1"
    name="example"
    type="radio"
    value="test1"
  />
  <label
    className="slds-nav-vertical__action"
    htmlFor="test1"
  >
    <span
      className="slds-nav-vertical_radio-faux"
    />
  </label>
</span>
`;

exports[`Vertical Nav Radio Group renders a shaded fieldset 1`] = `
<fieldset
  className="slds-nav-vertical slds-nav-vertical_shade"
>
  <legend
    className="slds-nav-vertical__title slds-text-title_caps"
  >
    Reports
  </legend>
  <span
    className="slds-nav-vertical__item"
  >
    <input
      defaultChecked={undefined}
      id="test1"
      name="example"
      type="radio"
      value="test1"
    />
    <label
      className="slds-nav-vertical__action"
      htmlFor="test1"
    >
      <span
        className="slds-nav-vertical_radio-faux"
      >
        test1
      </span>
    </label>
  </span>
</fieldset>
`;
